crates/subnetworks/history/src/storage.rs

```diff
@@ -35,6 +36,12 @@ impl ContentStore for HistoryStorage {
         key: &HistoryContentKey,
     ) -> Result<ShouldWeStoreContent, ContentStoreError> {
         let content_id = ContentId::from(key.content_id());
+        // temporarily disable storing all block headers
+        if self.disable_storing_headers {
+            if let HistoryContentKey::BlockHeaderByHash(_) = key {
+                return Ok(ShouldWeStoreContent::NotWithinRadius);
+            }
+        }
```

Comment:
```suggestion
        // temporarily disable storing all block headers
        if self.disable_storing_headers {
            if let HistoryContentKey::BlockHeaderByHash(_) | HistoryContentKey::BlockHeaderByNumber(_) = key {
                return Ok(ShouldWeStoreContent::NotWithinRadius);
            }
        }
```

Don't you need to add a check for BlockHeaderByNumber as well